File: jung-io/src/main/java/edu/uci/ics/jung/io/graphml/AbstractMetadata.java
/*
 * Copyright (c) 2008, The JUNG Authors
 *
 * All rights reserved.
 *
 * This software is open-source under the BSD license; see either
 * "license.txt" or
 * https://github.com/jrtom/jung/blob/master/LICENSE for a description.
 */

package edu.uci.ics.jung.io.graphml;

import java.util.HashMap;
import java.util.Map;

/**
 * Abstract base class for metadata - implements the property functionality
 * 
 * @author Nathan Mittler - nathan.mittler@gmail.com
 */
public abstract class AbstractMetadata implements Metadata {

    final private Map<String,String> properties = new HashMap<String, String>();
    
    public Map<String, String> getProperties() {
        return properties;
    }
    
    public String getProperty(String key) {
        return properties.get(key);
    }
    
    public String setProperty(String key, String value) {
        return properties.put(key, value);
    }
    
    public void addData(DataMetadata data) {
        properties.put(data.getKey(), data.getValue());
    }
}

File: jung-io/src/main/kotlin/edu/uci/ics/jung/io/graphml/AbstractMetadata.kt
/*
 * Copyright (c) 2008, The JUNG Authors
 *
 * All rights reserved.
 *
 * This software is open-source under the BSD license; see either
 * "license.txt" or
 * https://github.com/jrtom/jung/blob/master/LICENSE for a description.
 */
package edu.uci.ics.jung.io.graphml

/**
 * Abstract base class for metadata - implements the property functionality
 *
 * @author Nathan Mittler - nathan.mittler@gmail.com
 */
abstract class AbstractMetadata : Metadata {

  override val properties: MutableMap<String, String> = HashMap()

  fun getProperty(key: String): String? = properties[key]

  fun setProperty(key: String, value: String): String? = properties.put(key, value)

  fun addData(data: DataMetadata) {
    properties[data.key!!] = data.value!!
  }
}
